.github/workflows/pr_checks.yml

```diff
@@ -0,0 +1,156 @@
+name: PR Checks
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths: # Only run tests when certain paths change
+      - fidesctl/**
+      - .github/workflows/test.yaml
+
+env:
+  CONTAINER: fidesctl-${{ github.event.pull_request.head.sha }}
+  IMAGE: ethyca/fidesctl:${{ github.event.pull_request.head.sha }}
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build fidesctl container
+        uses: docker/build-push-action@v2
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          context: ./fidesctl
+          file: ./fidesctl/Dockerfile
+          outputs: type=docker,dest=/tmp/${{ env.CONTAINER }}.tar
+          push: false
+          tags: ${{ env.IMAGE }}
+
+      - name: Upload fidesctl container
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.CONTAINER }}
+          path: /tmp/${{ env.CONTAINER }}.tar
+          retention-days: 7
+
+  Evaluate:
+    needs: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download fidesctl container
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ env.CONTAINER }}
+          path: /tmp/
+
+      - name: Load fidesctl image
+        run: docker load --input /tmp/${{ env.CONTAINER }}.tar
```

Comment:
If we want to DRY things up further, we could create a separate action for these two tasks. It didn't seem absolutely necessary yet.